src/main/java/org/project36/qualopt/config/social/SocialConfiguration.java

```diff
@@ -95,8 +100,36 @@ public void addConnectionFactories(ConnectionFactoryConfigurer connectionFactory
         }
 
         // Twitter configuration
-        String twitterClientId = environment.getProperty("spring.social.twitter.client-id");
-        String twitterClientSecret = environment.getProperty("spring.social.twitter.client-secret");
+        String twitter_config_file_path = "src\\main\\resources\\config\\.twitter_config.properties";
+        this.configure_Twitter(connectionFactoryConfigurer, twitter_config_file_path);
+
+        // jhipster-needle-add-social-connection-factory
+    }
+
+    /*
+    This private method uses the hidden configuration file for Twitter to set up Twitter social connection.
+    The hidden configuration file for Twitter should be stored in "src\\main\\resources\\config\\.twitter_config.properties".
+     */
+    private void configure_Twitter(ConnectionFactoryConfigurer connectionFactoryConfigurer, String twitter_config_file_path) {
+        Properties prop = new Properties();
+        InputStream input = null;
+
+        String twitterClientId = null;
+        String twitterClientSecret = null;
+
+        try {
+            input = new FileInputStream(twitter_config_file_path);
+
+            // load a properties file
+            prop.load(input);
+
+            // get the property value and print it out
+            twitterClientId = prop.getProperty("twitter-client-id");
+            twitterClientSecret = prop.getProperty("twitter-client-secret");
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
```

Comment:
And actually FileInputStreams should be closed as a good practice?

Reply:
Hey,
The code has been updated to close the FileInputStream.

Reply:
Thanks, please investigate on the bug :)

Reply:
and maybe we could also log the exception? the Logger is already present in this class.